runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -87,7 +88,7 @@ protected ILogData removeFromQueue(NavigableSet<Long> queue) {
                     // Validate that the data entry belongs to this stream, otherwise, skip.
                     // This verification protects from sequencer regression (tokens assigned in an older epoch
                     // that were never written to, and reassigned on a newer epoch)
-                    if (ld.containsStream(this.id)) {
+                    if (ld.containsStream(this.id) && ld.getType() == DataType.DATA) {
```

Comment:
Should you just left the stream adapter to filter this out ?

Reply:
Actually, I came to this because a unit test was failing. If I'm not wrong it was a raw access to the stream and it just got stuck infinitely if we do not filter on the data type at this layer.